### PR TITLE
feat(mbb): make display_name_to_roster_key canonical in sdv-py

### PR DESCRIPTION
**Pull request by saiemgilani**
Box-score and shot-chart pages render a player `"Surname, First"`; `team_rosters` renders the same person `FIRST.MIDDLE.LAST` uppercase. Joining them needs one canonical direction — and it was living in **two copies** inside the `-data` repos' `ops/publish_rapm.py`: NCAA name-matching domain logic sitting outside the module that owns NCAA name matching, untested, with two places to drift.

Now exported from `mbb_ncaa_names` alongside `code_from_box` and `tidy_player`.

Each normalization is documented with the match rate that earned it, measured on real 2024 MBB data:

| step | match rate |
|---|---|
| naive comma split | 93.04% |
| + suffix / quoted-nickname strip | 98.07% |
| + whitespace → dots (multi-token surnames) | 99.08% |

The last step is the subtle one: the roster keeps **interior dots as token separators**, so collapsing `"Ballisager Webb"` to `BALLISAGERWEBB` silently misses.

An unsplittable name returns `""` rather than a partial key — an empty key never matches, and an unresolved row beats a wrong join.

12 new tests (48 in the file). Follow-up: point both `-data` publishers at this once the twins re-lock their sdv-py pin.

## Summary by Sourcery

Provide one canonical, fail-closed conversion from NCAA display names to roster keys for reliable player joins.

New Features:
- Add a canonical utility for converting display-formatted NCAA player names into roster-style join keys.
- Expose the name-normalization utility through the MBB and WBB public APIs.

Bug Fixes:
- Prevent incomplete or unsplittable names from producing misleading partial join keys.

Enhancements:
- Centralize NCAA display-name normalization, including suffixes, nicknames, multi-token surnames, hyphens, and diacritics.

Documentation:
- Document the new canonical name-conversion utility and its roster-joining behavior for MBB and WBB.

Tests:
- Add coverage for standard, formatted, multi-token, invalid, and fail-closed player names, including the WBB re-export.

## Summary by CodeRabbit

* **New Features**
  * Added a utility that converts display names into consistent roster keys.
  * Supports nicknames, suffixes, punctuation, hyphens, whitespace, and accented characters.
  * Available across men’s and women’s basketball namespaces, including legacy access paths.

* **Documentation**
  * Added reference documentation and updated function counts for basketball modules.

* **Tests**
  * Added coverage for standard names, edge cases, invalid input, and cross-module availability.

### CODE REVIEW

File: sportsdataverse/mbb/mbb_ncaa_names.py [hunk 1]
```diff
@@ -191,6 +193,69 @@
...
+        return ""
+    name = _ROSTER_KEY_NICKNAME.sub(" ", name)
+    parts = [p.strip() for p in name.split(",")]
+    parts = [p for p in parts if p and not _ROSTER_KEY_SUFFIX.fullmatch(p)]
+    if len(parts) < 2:
+        return ""
+    first, last = parts[-1], " ".join(parts[:-1])
+    toks = [_roster_key_token(t) for t in (first.split() + last.split())]
+    toks = [t for t in toks if t and not _ROSTER_KEY_SUFFIX.fullmatch(t)]
+    return ".".join(toks)
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/mbb/mbb_ncaa_names.py]: Fixed, and the framing in the finding is the right one: this violated the documented unsplittable-name contract. I was filtering the COMBINED token list, so when the given-name field held nothing usable the surname alone survived and got joined.

```
'Smith, Jr. III' -> ''   (was 'SMITH')
'Smith, 123'     -> ''   (was 'SMITH')
'123, Smith'     -> ''
'Clark, Garry'   -> 'GARRY.CLARK'   (unchanged)
```

Both components must now survive normalization or the key is empty. The reason this matters more than a normal edge case: an empty key never matches, but `SMITH` JOINS — to whoever else on that roster reduces to `SMITH`. A partial key is worse than no key.

[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/mbb/mbb_ncaa_names.py]: Fixed, and the framing in the finding is the right one: this violated the documented unsplittable-name contract. I was filtering the COMBINED token list, so when the given-name field held nothing usable the surname alone survived and got joined.

```
'Smith, Jr. III' -> ''   (was 'SMITH')
'Smith, 123'     -> ''   (was 'SMITH')
'123, Smith'     -> ''
'Clark, Garry'   -> 'GARRY.CLARK'   (unchanged)
```

Both components must now survive normalization or the key is empty. The reason this matters more than a normal edge case: an empty key never matches, but `SMITH` JOINS — to whoever else on that roster reduces to `SMITH`. A partial key is worse than no key.

File: sportsdataverse/mbb/mbb_ncaa_names.py [hunk 2]
```diff
@@ -156,6 +157,7 @@
     "build_tidy_player_context",
     "tidy_player",
     "code_from_box",
+    "display_name_to_roster_key",
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/mbb/mbb_ncaa_names.py:160]: Fixed. The shim documents itself as re-exporting the shared NCAA name functions, so a name added to mbb and not to wbb makes that documentation false — `from sportsdataverse.wbb.wbb_ncaa_names import display_name_to_roster_key` would fail while the module claims otherwise.

Added to both the import list and `__all__`, with a test that pins each separately (importable AND declared), since exporting it while forgetting `__all__` is the same silent gap one layer down.

File: tests/mbb/test_mbb_shots_period.py [hunk 3]
```diff
@@ -136,3 +136,45 @@ class TestNonFiniteClock:
...
+
+        assert display_name_to_roster_key(bad) == ""  # type: ignore[arg-type]
+
+    def test_multi_token_surname_is_not_concatenated(self) -> None:
+        """The subtle one: rosters keep INTERIOR dots as token separators."""
+        from sportsdataverse.mbb.mbb_ncaa_names import display_name_to_roster_key
+
+        got = display_name_to_roster_key("Tchamwa Tchatchoua, Jonathan")
+        assert got == "JONATHAN.TCHAMWA.TCHATCHOUA"
+        assert got != "JONATHAN.TCHAMWATCHATCHOUA"
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tests/mbb/test_mbb_shots_period.py]: Fixed — moved to `tests/mbb/test_mbb_ncaa_names.py`. They cover `mbb_ncaa_names`, not the shots adapter they were sitting beside; I'd put them there only because that file was open. The layout convention is right and `test_<sport>_<module>.py` now matches.
